src/com/google/javascript/jscomp/RewriteDynamicImports.java

```diff
@@ -0,0 +1,312 @@
+        moduleNamespaceType = registry.getNativeType(JSTypeNative.UNKNOWN_TYPE);
+      }
+      promiseDotResolve.setJSType(
+          registry.createFunctionType(
+              registry.createTemplatizedType(
+                  registry.getNativeObjectType(JSTypeNative.PROMISE_TYPE), moduleNamespaceType),
+              registry.getNativeType(JSTypeNative.ALL_TYPE)));
+      promiseResolveCall.addChildToBack(createModuleNamespaceNode(targetModuleNS));
+      promiseResolveCall.setJSType(dynamicImport.getJSType());
+    } else {
+      promiseDotResolve.setJSType(
+          registry.createFunctionType(
+              registry.createTemplatizedType(
+                  registry.getNativeObjectType(JSTypeNative.PROMISE_TYPE),
+                  registry.getNativeType(JSTypeNative.VOID_TYPE)),
+              registry.getNativeType(JSTypeNative.ALL_TYPE)));
+      promiseResolveCall.setJSType(
+          registry.createTemplatizedType(
+              registry.getNativeObjectType(JSTypeNative.PROMISE_TYPE),
+              registry.getNativeType(JSTypeNative.VOID_TYPE)));
+    }
+    promiseResolveCall.srcrefTree(dynamicImport);
+    Node parent = dynamicImport.getParent();
+    dynamicImport.replaceWith(promiseResolveCall);
+    t.reportCodeChange(parent);
+    dynamicImportsRemoved = true;
+  }
+
+  private static String getChunkFileName(JSModule chunk) {
+    return chunk.getName() + ".js";
+  }
+
+  /**
+   * Rewrite the dynamic import specifier to the output chunk path
+   *
+   * <p>Before
+   *
+   * <pre>
+   *   import('./foo.js');
+   * </pre>
+   *
+   * After
+   *
+   * <pre>
+   *   import('./chunk0.js');
+   * </pre>
+   */
+  private void retargetImportSpecifier(NodeTraversal t, Node dynamicImport, JSModule targetModule) {
+    String retargetedSpecifier;
+    String importingChunkFilename = getChunkFileName(t.getInput().getModule());
+    String targetChunkFilename = getChunkFileName(targetModule);
+    try {
+      retargetedSpecifier =
+          ModuleLoader.relativePathFrom(importingChunkFilename, targetChunkFilename);
+    } catch (IllegalArgumentException e) {
+      compiler.report(
+          JSError.make(
+              dynamicImport,
+              UNABLE_TO_COMPUTE_RELATIVE_PATH,
+              importingChunkFilename,
+              targetChunkFilename));
+      return;
+    }
+    final Node originalSpecifierNode = dynamicImport.getFirstChild();
+    Node newSpecifier =
+        astFactory.createString(retargetedSpecifier).srcref(originalSpecifierNode);
+    originalSpecifierNode.replaceWith(newSpecifier);
+    t.reportCodeChange(newSpecifier);
+  }
+
+  /**
+   * Since a dynamic import expression is a promise resolving to the namespace export type, add a
+   * ".then()" call after it and resolve to the rewritten module namespace.
+   *
+   * <p>Before
+   *
+   * <pre>
+   *   import('./foo.js');
+   * </pre>
+   *
+   * After
+   *
+   * <pre>
+   *   import('./foo.js').then(function() { return module$foo; });
```

Comment:
This code actually adds an arrow function.
Could you update this comment to match?

Reply:
Done